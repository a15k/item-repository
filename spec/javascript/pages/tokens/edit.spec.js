import Edit from 'pages/tokens/editing';
import Factory from '../../factories';
import renderer from 'react-test-renderer';
jest.mock('components/surety-guard', () => ({ onConfirm, children }) => (
  <span className="surety-guard" onClick={onConfirm}>{children}</span>
));


describe(Edit, () => {
  let props;
  let tokens;

  beforeEach(() => {
    tokens = Factory.accessTokenCollection({ count: 1 });
    props = {
      tokens,
      isActive: true,
      onDone: jest.fn(),
      token: tokens.array[0],
    };
  });

  it('matches snapshot', () => {
    const edit = renderer.create(<Edit {...props} />);
    expect(edit.toJSON()).toMatchSnapshot();
    edit.unmount();
  });

  it('calls delete', () => {
    tokens.destroy = jest.fn(() => Promise.resolve());
    const edit = mount(<Edit {...props} />);
    edit.find('Button[icon="trash"]').simulate('click');
    expect(tokens.destroy).toHaveBeenCalled();
    edit.unmount();
  });

  it('displays and edits values', () => {
    fetch.mockResponseOnce(JSON.stringify({
      data: { name: 'my app', token: '1234' },
    }));

    const edit = mount(<Edit {...props} />);
    expect(edit).toRender(`input[defaultValue="${props.token.name}"]`);
    edit.find('[name="name"][className="form-control"]').getDOMNode().value = 'my token';
    edit.find('Button[icon="save"]').simulate('click');
    expect(fetch.mock.calls.length).toEqual(1);
    expect(JSON.parse(fetch.mock.calls[0][1].body)).toMatchObject({
      name: 'my token',
    });
    expect(props.token.name).toContain('my token');
    edit.unmount();
  });


});
