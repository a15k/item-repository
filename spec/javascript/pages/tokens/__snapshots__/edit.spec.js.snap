// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokensEdit matches snapshot 1`] = `
<form
  className="sc-htpNat hbzTHP"
>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="name"
    >
      Name
    </label>
    <input
      className="form-control"
      defaultValue="Movies"
      id="name"
      name="name"
      type="text"
    />
  </div>
  <div
    className="token"
  >
    <h3>
      Token
    </h3>
    <textarea
      readOnly={true}
      value="8700c4e3-5b9e-41f5-9e6a-768cc5796d96-fc028b7b-635e-4d7a-a00f-2b0eae9f8604-865c8911-9b92-4009-9e7c-1f6f1ce4c967-03e6c097-3f41-4a1e-ba95-731031eb6f8b"
    />
  </div>
  <div
    className="sc-bwzfXH cryUfd"
  >
    <button
      className="sc-bdVaJa gZEDXH btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-chevron-left fa-w-10 icon"
        data-icon="chevron-left"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Cancel
      </span>
    </button>
    <span
      className="surety-guard"
      onClick={[Function]}
    >
      <button
        className="sc-bdVaJa kaVmVf btn btn-secondary"
        onClick={[Function]}
        type={undefined}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash-alt fa-w-14 icon"
          data-icon="trash-alt"
          data-prefix="fas"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 84V56c0-13.3 10.7-24 24-24h112l9.4-18.7c4-8.2 12.3-13.3 21.4-13.3h114.3c9.1 0 17.4 5.1 21.5 13.3L312 32h112c13.3 0 24 10.7 24 24v28c0 6.6-5.4 12-12 12H12C5.4 96 0 90.6 0 84zm416 56v324c0 26.5-21.5 48-48 48H80c-26.5 0-48-21.5-48-48V140c0-6.6 5.4-12 12-12h360c6.6 0 12 5.4 12 12zm-272 68c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208zm96 0c0-8.8-7.2-16-16-16s-16 7.2-16 16v224c0 8.8 7.2 16 16 16s16-7.2 16-16V208z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span>
          Delete
        </span>
      </button>
    </span>
    <button
      className="sc-bdVaJa gZEDXH btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-save fa-w-14 icon"
        data-icon="save"
        data-prefix="fas"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M433.941 129.941l-83.882-83.882A48 48 0 0 0 316.118 32H48C21.49 32 0 53.49 0 80v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V163.882a48 48 0 0 0-14.059-33.941zM224 416c-35.346 0-64-28.654-64-64 0-35.346 28.654-64 64-64s64 28.654 64 64c0 35.346-28.654 64-64 64zm96-304.52V212c0 6.627-5.373 12-12 12H76c-6.627 0-12-5.373-12-12V108c0-6.627 5.373-12 12-12h228.52c3.183 0 6.235 1.264 8.485 3.515l3.48 3.48A11.996 11.996 0 0 1 320 111.48z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Save
      </span>
    </button>
  </div>
</form>
`;
